### PR TITLE
AFP Bid Adapter: add new bid adapter and integration examples

### PR DESCRIPTION
**Pull request by andrey-ka-97**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- The name of the adapter has been changed;
- The adapter has been completely redesigned for the new api.

- test parameters for validating bids
```
var adUnits = [{
  code: 'iib-target',
  mediaTypes: {
    banner: {
      sizes: [[0, 0]],
    }
  },
  bids: [{
    bidder: "afp",
    params: {
      placeType: "In-image", // or "In-image Max"
      placeId: "60e7039918047b3fae304850",
      placeContainer: '#iib-container',
      imageUrl: "https://rtbinsight.ru/content/images/size/w1000/2021/05/ximage-30.png.pagespeed.ic.IfuX4zAEPP.png",
      imageWidth: 1000,
      imageHeight: 524,
    }
  }]
}];

var adUnits = [{
  code: 'iib-target',
  mediaTypes: {
    banner: {
      sizes: [[0, 0]],
    }
  },
  bids: [{
    bidder: "afp",
    params: {
      placeType: "In-content Banner", // or "In-content Stories" or "Action Scroller" or "Action Scroller Light"
      placeId: "60e7039918047b3fae304850",
      placeContainer: '#iib-container',
    }
  }]
}];

var adUnits = [{
  code: 'iib-target',
  mediaTypes: {
    video: {
      playerSize: [[480, 320]],
    }
  },
  bids: [{
    bidder: "afp",
    params: {
      placeType: "In-content Video", // or "Out-content Video"
      placeId: "60e7039918047b3fae304850",
      placeContainer: '#iib-container',
    }
  }]
}];
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- - andrey.kara.97@mail.ru
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
- - https://github.com/prebid/prebid.github.io/pull/3205

### CODE REVIEW

**Comment by patmmccann** [comment]
Deleting the astraone adapter is considered breaking and you'll have to make that change against the 6.0 branch that has not yet been created. Can you remove that part of the Pr for now?

**Review by bretg — COMMENTED** [review]
The name change is fine of course, but we recommend implementing an alias for the old 'astraone' name. We generally don't completely remove modules except at major releases in case publishers are using it. By implementing astraone as an alias, pubs can continue using the original name. 

However, I do note from the docs PR that the params are different. So if you're absolutely sure that you want to remove the astraone name permanently from 5.x, ok.

**Comment by andrey-ka-97** [comment]
revert old adapter.
and also here https://github.com/prebid/prebid.github.io/pull/3205

**Comment by andrey-ka-97** [comment]
What's wrong with the tests? I ran the command "gulp test --file "test/spec/modules/afpBidAdapter_spec.js"" and they completed successfully.

**Comment by bretg** [comment]
@ChrisHuie is this build issue happening for others? Perhaps this fork needs to be updated with a fix from another adapter?

**Comment by ChrisHuie** [comment]
@andrey-ka-97 rerunning tests. Seemed to have a weird IE disconnect from bowserstack. There are a bunch of integration tests and tests on storage and other modules that are all ran in CircleCI. So we sometimes get flaky test fails

File: test/spec/modules/afpBidAdapter_spec.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  return {
+    params: params,
+    sizes,
+    bidId,
+    bidder: BIDDER_CODE,
+    mediaTypes: mediaTypes,
+    transactionId,
+  }
+}
+const validBidRequests = Object.values(configByPlaceType).map(config => getTransformedConfig(config))
```
Comment (ChrisHuie) [inline test/spec/modules/afpBidAdapter_spec.js:121]: I think this is the issue with IE? Object.values isn't supported in IE (https://caniuse.com/object-values). You will need to refactor this using object.keys or a different method

**Review by ChrisHuie — CHANGES_REQUESTED** [review]

File: test/spec/modules/afpBidAdapter_spec.js (hunk 1 above)
Comment (andrey-ka-97) [reply test/spec/modules/afpBidAdapter_spec.js:121]: These are tests, they will not be executed in IE. Moreover, gulp, when building production, must convert the code to the old standard.
There is not only an object.values, there are also "entries", as well as destructurization and dynamic keys for objects. But these are just tests and they will not get into the "prebid.js" assembly in the browser.

**Comment by andrey-ka-97** [comment]
so what is the bottom line? Is it a normal situation that in the file with tests (afpBidAdapter_spec.js) I need to write in the old ES standard?

**Comment by andrey-ka-97** [comment]
@bretg now there are no errors, when will the adapter be merged with the master branch?

**Comment by robertrmartinez** [comment]
@andrey-ka-97 

This was assigned to me and I have been out for a couple weeks.

I will give this a thorough review this in the next day or so.

**Comment by andrey-ka-97** [comment]
> @andrey-ka-97
> 
> This was assigned to me and I have been out for a couple weeks.
> 
> I will give this a thorough review this in the next day or so.

**Comment by andrey-ka-97** [comment]
@robertrmartinez please tell me when will you check the adapter?

**Comment by robertrmartinez** [comment]
@andrey-ka-97 

So a publisher is required to load some JS or add it themselves on the `afp` object on the window?

Is this part of your onboarding with pubs where you have them load some AFP script?

**Comment by andrey-ka-97** [comment]
@robertrmartinez 
"afp" is an appeal to our "sdk". And the " sdk "itself will be loaded into an "iframe" and work with the publisher's page, as our formats require it.

**Comment by robertrmartinez** [comment]
So this is only meant to be used when in conjunction with the SDK?

Think that should be noted in the .md and Docs files if so.

**Comment by robertrmartinez** [comment]
@andrey-ka-97 I am not able to solicit bits using the input params.

The afp endpoint gives me a CORS error every time.

Bid Adapters are supposed to have valid test params so we can consistently get test bid responses to make sure integration works.

Can you please update the .md / docs with how to achieve this please!

Thanks

**Comment by andrey-ka-97** [comment]
@robertrmartinez 
the problem with CORS is solved. 
as for the SDK, the publisher does not need to add it to his page on his own. My adapter will do it for him and therefore in the files .md probably shouldn't report this.

**Comment by robertrmartinez** [comment]
@andrey-ka-97 

I got your `In-image` ad to work

So this ad requires to hook on and display over an image, it looks pretty cool.

Sorry for my lack of SDK knowledge but is this common? Are other prebid bidders set to actually run in same auction with same slots as your placements?

I tried to get `In-content Banner` ad to work, which sounds more of the usual use case, but could not. The endpoint returns no-bids.

I want to run your adapter alongside another adapter in a basic banner render scenario and make sure it works as expected.

**Comment by andrey-ka-97** [comment]
@robertrmartinez 

The SDK is a development of our company, it is used by us to visualize different types of formats.

As a regular banner with fixed sizes, you need to use the "Just Banner" format.

The "In-image" format is also used in "voxBidAdapter".

**Review by robertrmartinez — APPROVED** [review]
ok makes sense

Thanks for your patience @andrey-ka-97

**Comment by andrey-ka-97** [comment]
@robertrmartinez 

please tell me, I see our adapter here (https://docs.prebid.org/download.html) and I choose the latest version (5.14), but an error occurs during the build ({"error": "Prebid file not built properly","RequestId": "ac0b6981-e328-488a-901f-d123cc9747dd"}), is it possible that our adapter is missing in the current latest version and will be added in the next version? if so, when will the next version with our adapter be released?

**Comment by bretg** [comment]
AFP will be in 5.15 which will be Weds or Thurs -- https://github.com/prebid/Prebid.js/releases

Sorry - I shouldn't have merged the docs before the release goes out

**Pull request merged**